OsmAnd/src/net/osmand/plus/plugins/externalsensors/adapters/DevicesForWidgetAdapter.java

```diff
@@ -0,0 +1,75 @@
+package net.osmand.plus.plugins.externalsensors.adapters;
+
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.plugins.externalsensors.devices.AbstractDevice;
+import net.osmand.plus.plugins.externalsensors.viewholders.FoundDeviceViewHolder;
+import net.osmand.plus.utils.UiUtilities;
+
+public class DevicesForWidgetAdapter extends FoundDevicesAdapter {
+
+	private final SelectDeviceListener deviceClickListener;
+	private UiUtilities uiUtils;
+	private String selectedDeviceId;
+	private int selectedPosition = -1;
```

Comment:
Is it necessary to store selectedPosition based on selectionMark for UI update? Will it introduce bugs with items update?
I suppose something [like this](https://github.com/osmandapp/OsmAnd/blob/aafb537c9f14907e3dce47619dcc95e7c9d11645/OsmAnd/src/net/osmand/plus/mapcontextmenu/editors/PointEditorFragment.java#L478-L482) will be more reliable

Reply:
i think, getting item position is heavier than just storing 1 int param